[Telink] Add Thread Optimization Configs

damien0x0023 opened the pull request:
- OPENTHREAD_CSMABACKOFF_OPTIMIZATION
- IEEE802154_TLX_OPTIMIZATION

and set COMMON_LIBC_MALLOC_ARENA_SIZE to 20kb if tl7218x_retention for power consumption optimization based on d6882ec0719692a1ef1bc983e138f32bf1a39442.

- The first commit (f5342e56dcf282a76d3dcec552ab7e006bf032d9) is only suitable for Zephyr V3.3 ( https://github.com/telink-semi/zephyr/pull/488 - passed and merged to develop_v3.3 - Note that I did manually squash so the actual commit hash is [86fcdb43b64d6d10ad81172c61495c9e40de0403 ](https://github.com/telink-semi/zephyr/tree/develop_v3.3)  )

- The second commit (61263b0304c1508491181760d53d405f824d029b) does not work for tl7218x_retention as it cannot complete commissioning due to the limit of LIBC allocation size 16kB.

- The third commit (dfabb0e35f6f13cf40bfc1bd3eeb9f8d825345d5) increases the limit to 20kB for tl7218x_retention while it maintains 16kB for tl7218kB so it is compatible for both Zephyr V3.3 and V3.7 (https://github.com/telink-semi/zephyr/pull/491 - passed and merged to develop)

- The fourth commit (1cf5186db505e2c9185986e6394504134c87179e) updates the description of the newly added configs and the Zephyr V3.7 revision after https://github.com/telink-semi/zephyr/pull/491 is merged.

#### Testing
Verified by CI

damien0x0023 committed:
[Telink] Add Thread Optimization Configs

- OPENTHREAD_CSMABACKOFF_OPTIMIZATION
- IEEE802154_TLX_OPTIMIZATION

and set COMMON_LIBC_MALLOC_ARENA_SIZE to 20kb if tl7218x_retention
for power consumption optimization
based on d6882ec0719692a1ef1bc983e138f32bf1a39442.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

haiwentelink commented on .github/workflows/examples-telink.yaml:
```diff
@@ -179,18 +179,18 @@ jobs:
-            #         telink tl3218x_retention light-switch-app-ota-shell-factory-data \
-            #         out/telink-tl3218x_retention-light-switch-ota-shell-factory-data/zephyr/zephyr.elf \
-            #         /tmp/bloat_reports/
+            - name: Build example Telink (tl321x_retention) Light Switch App with OTA, Shell, Factory Data
```
need to delete , or zephyr3.7 wil fail .temporary check is ok, later can delete.

damien0x0023 committed:
[Telink] Trigger CI for zephyr v3.7

- update zephyr revision to zephyr V3.7
- comment tl3218x_retention example due to ram limit.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

damien0x0023 committed:
[Telink] Increase LIBC size for tl7218x_retention

- increase COMMON_LIBC_MALLOC_ARENA_SIZE from 16kB to 20kB
for tl7218x_retention to avoid overflow issue.
- improve configs description.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

damien0x0023 committed:
[Telink] Update config description and zephyr revision

- update OPENTHREAD_CSMABACKOFF_OPTIMIZATION description
- update zephyr revision

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

damien0x0023 commented:
@s07641069 @haiwentelink I believe it's okay to merge after your review.
